portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -19,14 +19,36 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  private ArrayList<String> data;
+
+  @Override
+  public void init() {
+    data = new ArrayList<String>();
+    data.add("Malcolm from NC");
+    data.add("Sabrina from FL");
+    data.add("Dan from TX");
+  }
+
+  private String convertToJson(ArrayList<String> messages) {
+    String json = "{ \"messages\" : [\"";
+    json += String.join("\",\"",messages);
+    json += "\"] }";
+    return json;
+  }
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    // Convert messages to JSON
+    String json = convertToJson(data);
+
+    // Send JSON as the response
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
   }
-}
+}
```

Comment:
nit: in general it is recommended to have a new line character (an empty new line) at the end of each file. This will make git diff works better.